algopy_testing/src/algopy_testing/primitives/string.py

```diff
@@ -12,10 +12,11 @@ class String:
     indexing and length operations are not supported. Use .bytes.length for byte length.
     """
 
-    value: str
+    _value: bytes  # underlying 'bytes' value representing the String
 
     def __init__(self, value: str = "") -> None:
-        self.value = as_string(value)
+        value = as_string(value)
```

Comment:
Unfortunately can't use as_string here as this should strictly only be the `str` type, `as_string` would also allow `algopy.String` and `algopy.arc4.String`

Reply:
It has been fixed in this commit 9e8f1dfaa2b440b9ef3241ad3290abf7df87bce9.